packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -136,21 +139,28 @@ class Checkbox extends StatefulWidget {
   /// state, it will be used instead of this color.
   final Color? activeColor;
 
+  /// {@template flutter.material.checkbox.fillColor}
   /// The color that fills the checkbox when it is checked, in all
   /// [MaterialState]s.
   ///
-  /// If this is provided, it will be used over [activeColor].
-  ///
   /// Resolves in the following states:
   ///  * [MaterialState.selected].
   ///  * [MaterialState.hovered].
   ///  * [MaterialState.focused].
   ///  * [MaterialState.disabled].
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state. If that is also null, the value of [CheckboxThemeData.fillColor]
+  /// is used.
```

Comment:
We're not explaining the component defaults here. It's a shame that the defaults aren't (yet) defined in terms of the theme's colorScheme however for the sake of consistency we should reveal what the defaults are.

Reply:
I did this now for Checkbox, Radio and Switch. For `Switch` it ended up being quite complicated to document the default values so I made a table for them. Hopefully that is easy to read.

Reply:
It's easy to read and I appreciate the trouble you've gone to to make it correct. 

Generally speaking, we want the default values of component color properties to be based on the theme's colorScheme, obviously there's still much work to do. That will have to wait for another PR.